crowbar_framework/lib/crowbar/upgrade_timeouts.rb

```diff
@@ -0,0 +1,37 @@
+require "yaml"
+
+module Crowbar
+  class UpgradeTimeouts
+    def values
+      @timeouts_config = begin
+        YAML.load_file("/etc/crowbar/upgrade_timeouts.yml")
```

Comment:
Hm, you're instantiating the UpgradeTimeouts objects in the ```Api::Node``` and ```Api::Upgrade``` constructors. AFAIU these get recreated with every API request, right? So it should be possible to load that file only once per API request at least.
But I really think that any optimization is currently not needed here.